packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -138,73 +152,49 @@ class CupertinoAlertDialog extends StatelessWidget {
   /// Typically this is a list of [CupertinoDialogAction] widgets.
   final List<Widget> actions;
 
-  /// A scroll controller that can be used to control the scrolling of the message
-  /// in the dialog.
+  /// A scroll controller that can be used to control the scrolling of the
+  /// message in the dialog.
   ///
-  /// Defaults to null, and is typically not needed, since most alert messages are short.
+  /// Defaults to null, and is typically not needed, since most alert messages
+  /// are short. Note that another scroll controller, [actionScrollController],
+  /// is needed for the actions section when there are many actions. So, it
+  /// would be nice to disambiguate this scroll controller by changing to
+  /// 'titleScrollController', TODO(ekbiker): change to "titleScrollController",
+  /// but it would be a breaking change.
   final ScrollController scrollController;
 
+  /// A scroll controller that can be used to control the scrolling of the
+  /// actions in the dialog.
+  ///
+  /// Defaults to null, and is typically not needed.
+  final ScrollController actionScrollController;
+
   @override
   Widget build(BuildContext context) {
-    const double edgePadding = 20.0;
-    final List<Widget> titleContentGroup = <Widget>[];
-    if (title != null) {
-      titleContentGroup.add(new Padding(
-        padding: new EdgeInsets.only(
-          left: edgePadding,
-          right: edgePadding,
-          bottom: content == null ? edgePadding : 1.0,
-          top: edgePadding,
-        ),
-        child: new DefaultTextStyle(
-          style: _kCupertinoDialogTitleStyle,
-          textAlign: TextAlign.center,
-          child: title,
-        ),
-      ));
-    }
-
-    if (content != null) {
-      titleContentGroup.add(
-        new Padding(
-          padding: new EdgeInsets.only(
-            left: edgePadding,
-            right: edgePadding,
-            bottom: edgePadding,
-            top: title == null ? edgePadding : 1.0,
-          ),
-          child: new DefaultTextStyle(
-            style: _kCupertinoDialogContentStyle,
-            textAlign: TextAlign.center,
-            child: content,
-          ),
-        ),
-      );
-    }
-
     final List<Widget> children = <Widget>[];
-    if (titleContentGroup.isNotEmpty) {
-      children.add(
-        new Flexible(
-          child: new CupertinoScrollbar(
-            child: new ListView(
-              controller: scrollController,
-              shrinkWrap: true,
-              children: titleContentGroup,
-            ),
-          ),
-        ),
-      );
-    }
 
-    if (actions != null) {
-      children.add(new _CupertinoButtonBar(
-        children: actions,
-      ));
+    final Widget titleSection = new _CupertinoAlertTitleSection(
+      title: title,
+      message: content,
+      scrollController: scrollController,
+    );
+    if (titleSection != null) {
```

Comment:
Ah, I see what you're saying. So you're checking null against the class's construction which will never return null. The widget's build method will be invoked separately by the framework. 

I'd encourage adding a test to exercise this path and the action section's path.